src/main/java/Duke.java

```diff
@@ -1,10 +1,147 @@
+            return "["+this.getStatusIcon()+ "] "+description;
+        }
+    }
+    public static class Todo extends Task{
+
+        public Todo(String description) {
+            super(description);
+        }
+        @Override
+        public String toString(){
+            return "[T]" + super.toString();
+        }
+
+    }
+    public static class Deadline extends Task{
+        protected String by;
+        public Deadline(String description) {
+            super(description);
+        }
+        public void setBy(String by ){
+            this.by = by;
+        }
+        @Override
+        public String toString() {
+            return "[D]" + super.toString() + " (by: " + this.by + ")";
+        }
+    }
+    public static class Event extends Task{
+        protected String at;
+        public Event(String description) {
+            super(description);
+        }
+        public void setAt(String at ){
+            this.at = at;
+        }
+        @Override
+        public String toString(){
+            return "[E]"+super.toString()+" (at: "+this.at+")";
+        }
+    }
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        printWelcomeMessage();
+        String command;
+        Task [] tasks = new Task[100];
+        int tasksTotal = 0;
+        Task taskToAdd = null;
+        Task taskToMark;
+        Scanner in = new Scanner(System.in);
+        command = in.nextLine();
+        while(!command.equals("bye")){
+            if(command.equals("list")) printList(tasks, tasksTotal);
+            else if(command.contains("done")){
+                int index = Integer.parseInt(command.substring("done".length()+1));
+                taskToMark = tasks[index-1];
+                taskToMark.markedAsDone();
+            }
+            else if(command.contains("todo")||command.contains("deadline")||command.contains("event")){
+                if (command.contains("todo")) {
+                    String taskTodo = command.substring("todo".length()+1);
+                    taskToAdd = new Todo(taskTodo);
+                }
+                if(command.contains("deadline")){
+                    int position = command.indexOf("/");
+                    String taskDeadline = command.substring("deadline".length()+1,position-1);
+                    String by = command.substring(position+"/by".length()+1);
+                    taskToAdd = new Deadline(taskDeadline);
+                    ((Deadline) taskToAdd).setBy(by);
+                }
+                if(command.contains("event")){
+                    int position = command.indexOf("/");
+                    String taskEvent = command.substring("event".length()+1,position-1);
```

Comment:
You might want to consider reducing code reuse. I noticed how `substring` is being used multiple times for event/deadline etc. You could implement a method for this.